### PR TITLE
#23479 edit inline name of the variant

### LINKED ISSUE #23479: [UI] Being able to edit Experiment Variant name
Currently is not possible. If you need to rename, you would need to delete a create a new Variant. 

Proposed: 

A edit pencil icon when the name is mouseovered, which displays the sidebar when clicked. 

Acceptance criteria:
* Being able to rename the variant 
* Automated tests

### PR DESCRIPTION
**Pull request by oidacra**
## Proposed Changes
Use the inline PrimeNg component to allow the name of the variant.

## Checklist
- [X] Tests
- [X]  Translations
- [ ]  Security Implications Contemplated (add notes if applicable)

## Additional Info
** any additional useful context or info **

## Screenshots

https://user-images.githubusercontent.com/1909643/209575844-0c3287b0-210e-4a13-8898-9c5986a9dafc.mov

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-categories/dot-categories-list/dot-categories-list.component.spec.ts [hunk 1]
```diff
@@ -26,17 +26,16 @@ import { InputTextModule } from 'primeng/inputtext';
...
 
 @Component({
     selector: 'dot-test-host-component',
-    template: `<dot-categories-list></dot-categories-list>`
+    template: ` <dot-categories-list></dot-categories-list>`
 })
 class TestHostComponent {}
 
-describe('DotCategoriesListingTableComponent', () => {
+xdescribe('DotCategoriesListingTableComponent', () => {
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-categories/dot-categories-list/dot-categories-list.component.spec.ts:38]: Is this test broken?

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-variants/dot-experiments-configuration-variants.component.html [hunk 2]
```diff
@@ -76,6 +87,38 @@ <h2 class="flex align-items-center gap-1 uppercase">
...
+                    </span>
+                </ng-template>
+                <ng-template pTemplate="content">
+                    <input #variantEdited [value]="variant.name" pInputText type="text" />
+                    <button
+                        class="p-button-rounded p-button-text"
+                        #button
+                        [loading]="button.disabled"
+                        (click)="
+                            editVariantName(variantEdited.value, variant); button.disabled = true
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-variants/dot-experiments-configuration-variants.component.html:108]: How about doing this in a function? To make it more readable. 

Also, can we explain why we need the `button.disabled = true`? For future reference.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-variants/dot-experiments-configuration-variants.component.ts [hunk 3]
```diff
@@ -47,9 +63,17 @@ export class DotExperimentsConfigurationVariantsComponent {
 
     @Input() stepStatus: StepStatus;
     @Input() variants: Variant[];
-
     @Output() sidebarStatusChanged = new EventEmitter<SidebarStatus>();
     @Output() delete = new EventEmitter<Variant>();
+    @Output() edit = new EventEmitter<Pick<DotExperiment, 'name' | 'id'>>();
     @Output() save = new EventEmitter<Pick<DotExperiment, 'name'>>();
     @Output() goToEditPage = new EventEmitter<{ variant: Variant; mode: EditPageTabs }>();
+    @ViewChildren(Inplace) private inplaceInputs: QueryList<Inplace>;
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-variants/dot-experiments-configuration-variants.component.ts:71]: I don't see we are using this. Remove it if not needed unless it is a work in progress.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/dot-experiments-configuration.component.ts [hunk 4]
```diff
@@ -83,6 +88,19 @@ export class DotExperimentsConfigurationComponent implements OnInit {
...
+     * @param {Variant} variant
+     * @returns void
+     * @memberof DotExperimentsConfigurationComponent
+     */
+    editVariant(variant: Variant) {
+        this.dotExperimentsConfigurationStore.editVariant({
+            id: variant.id,
+            description: variant.name
+        });
+    }
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/dot-experiments-configuration.component.ts:102]: Should we test it?

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/store/dot-experiments-configuration-store.ts [hunk 5]
```diff
@@ -129,6 +139,42 @@ export class DotExperimentsConfigurationStore extends ComponentStore<DotExperime
         );
     });
 
+    readonly editVariant = this.effect(
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/store/dot-experiments-configuration-store.ts:142]: Should we test this?

**Review by rjvelazco — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-categories/dot-categories-list/dot-categories-list.component.spec.ts (hunk 1 above)
Comment (fmontes) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-categories/dot-categories-list/dot-categories-list.component.spec.ts:38]: Yes, it was old code I need it to merge.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/store/dot-experiments-configuration-store.ts [hunk 6]
```diff
@@ -129,6 +139,42 @@ export class DotExperimentsConfigurationStore extends ComponentStore<DotExperime
...
+        (variant$: Observable<{ id: string; description: string }>) => {
+            return variant$.pipe(
+                tap(() => this.setSidebarStatus(Status.SAVING)),
+                withLatestFrom(this.state$),
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/store/dot-experiments-configuration-store.ts:146]: Can't you pass the `experiment.id` from the component?

**Review by fmontes — CHANGES_REQUESTED** [review]

**Review by rjvelazco — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-variants/dot-experiments-configuration-variants.component.spec.ts [hunk 7]
```diff
@@ -89,7 +94,7 @@ describe('DotExperimentsConfigurationVariantsComponent', () => {
 
             spectator.setInput(variantsVm);
 
-            spectator.detectChanges();
+            spectator.detectComponentChanges();
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-variants/dot-experiments-configuration-variants.component.spec.ts:97]: oh!

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-variants/dot-experiments-configuration-variants.component.ts [hunk 8]
```diff
@@ -44,12 +60,22 @@ export class DotExperimentsConfigurationVariantsComponent {
...
+
+    constructor(private readonly cdr: ChangeDetectorRef) {}
+
+    editVariantName(newVariantName: string, variant: Variant, button: HTMLButtonElement) {
+        this.edit.emit({
+            ...variant,
+            name: newVariantName
+        });
+        button.disabled = true;
+        this.cdr.markForCheck();
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-variants/dot-experiments-configuration-variants.component.ts:79]: You need to the button attr in the template so you don't have to manually detect changes.

**Review by fmontes — CHANGES_REQUESTED** [review]

**Review by fmontes — APPROVED** [review]
Let's go!

**Pull request merged**